DevExpress/testcafe
# Add API for more flexible setting of "skip-js-error" option(closes #2775)

## Purpose
In some cases, it is necessary to specify more flexible parameters for the skipJsErrors option rather than the regular bool flag.

## API

```ts
interface SkipJsErrorsOptionsObject {
    stack?: string | RegExp;
    message?: string | RegExp;
    pageUrl?: string | RegExp;
}

type SkipJsErrorsCallback = (opts?: { message: string; stack: string; pageUrl: string }) => boolean;

interface SkipJsErrorsCallbackWithOptionsObject {
    fn: SkipJsErrorsCallback;
    dependencies?: { [key: string]: any };
}

//CLI and Configuration:
type SkipJsErrorsOptionValue = SkipJsErrorsOptionsObject | boolean;

//Runner
interface RunnerRunOptions {
    skipJsErrors?: SkipJsErrorsOptionValue | SkipJsErrorsCallback | SkipJsErrorsCallbackWithOptionsObject;
}

//TestController
interface TestController {
    /**
     * Set skipJsError options.
     *
     * @param options - Boolean flag, options object or callback function that should be used to skip client errors.
     */
    skipJsErrors (options?: boolean | SkipJsErrorsOptionsObject | SkipJsErrorsCallback | SkipJsErrorsCallbackWithOptionsObject): this;
}

//Fixture and Test methods:
interface FixtureFn {
    /**
     * Set skipJsError options.
     *
     * @param options - Boolean flag, options object or callback function that should be used to skip client errors.
     */
    skipJsErrors (options?: boolean | SkipJsErrorsOptionsObject | SkipJsErrorsCallback | SkipJsErrorsCallbackWithOptionsObject): this;
}

interface TestFn {
    /**
     * Set skipJsError options.
     *
     * @param options - Boolean flag, options object or callback function that should be used to skip client errors.
     */
    skipJsErrors (options?: boolean | SkipJsErrorsOptionsObject | SkipJsErrorsCallback | SkipJsErrorsCallbackWithOptionsObject): this;
}
```

Examples:
```js
target = Fixture | Test | TestController

target.skipJsErrors();
target.skipJsErrors(true);
target.skipJsErrors({ message: /someerr/i, stack: '/somestack/i', pageUrl:'https://example.com' });
target.skipJsErrors({ fn: ({ message, pageUrl, stack }) => message === ERROR_MESSAGE, dependencies: { ERROR_MESSAGE } });
target.skipJsErrors(({ message, pageUrl, stack }) => message === 'Some message');
```
CLI:
`testcafe chrome test.js -e message=testMessage,stack=testStack,pageUrl=testPageUrl`
